lib/util/persistent_memory_service.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:get_it/get_it.dart';
+import 'package:mazilon/util/logger_service.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+abstract class PersistentMemoryService {
+  Future<void> setItem(String key, String type, dynamic value);
+  Future<dynamic> getItem(
+    String key,
+    String type,
+  );
+  Future<void> reset();
+}
+
+class AndroidPersistentMemoryService implements PersistentMemoryService {
+  @override
+  Future<void> setItem(String key, String type, dynamic value) async {
+    IncidentLoggerService loggerService =
+        GetIt.instance<IncidentLoggerService>();
+    if (key == "" || value == null) {
+      loggerService.captureLog(
+        'Invalid key or value for persistent memory service',
+      );
+    }
+
+    try {
+      var prefs = await SharedPreferences.getInstance();
+      switch (type) {
```

Comment:
Better to have the 'string' 'int' etc types as a predefined list to avoid typos that we will find in runtime.
also - a defensive way would be to lower case the string before comparing to reduce the chance of typos even more (optionally if the list of types is predefined)